platform/linux-generic/odp_packet_io.c

```diff
@@ -1698,21 +1699,22 @@ int odp_pktin_recv_tmo(odp_pktin_queue_t queue, odp_packet_t packets[], int num,
 			if (odp_unlikely(!started)) {
 				odp_time_t t;
 
-				t = odp_time_local_from_ns(wait * SLEEP_NSEC);
+				t = odp_time_local_from_ns(wait * 1000);
 				started = 1;
 				t1 = odp_time_sum(odp_time_local(), t);
 			}
 
 			/* Check every SLEEP_CHECK rounds if total wait time
 			 * has been exceeded. */
-			if ((wait & (SLEEP_CHECK - 1)) == 0) {
-				t2 = odp_time_local();
+			for (i = 0; i < SLEEP_USEC; i++) {
+				if ((wait & (SLEEP_CHECK - 1)) == 0) {
+					t2 = odp_time_local();
 
-				if (odp_time_cmp(t2, t1) > 0)
-					return 0;
+					if (odp_time_cmp(t2, t1) > 0)
+						return 0;
+				}
+				wait--;
```

Comment:
Can roll over if SLEEP_USEC > 1.

Reply:
D'oh! That is a serious problem and definitely needs to be fixed.